src/python/WMCore/MicroService/MSTransferor/MSTransferor.py

```diff
@@ -676,3 +723,61 @@ def _diskPNNs(self, pnnList):
             else:
                 diskPNNs.add(pnn)
         return diskPNNs
+
+    def updateSites(self, rec):
+        """
+        Update sites API provides asynchronous update of site list information
+        :param rec: JSON payload with the following data structures: {'workflow': <wflow name>}
+        :return: either empty list (no errors) or list of errors
+        """
+        # preserve provided payload to local file system
+        wflowName = rec['workflow']
+        status = self.updateStorage(wflowName)
+        if status == 'ok':
+            return []
+        err = MSTransferorStorageError('storage error', **rec)
```

Comment:
How about adding a bit more context to this error message, like "Error persisting workflow in the storage"?

Reply:
Or probably better, use the `status` object returned from `updateStorage()` to have a more specific error message.

Reply:
ok